hazelcast/src/main/java/com/hazelcast/query/impl/predicates/PartitionPredicateImpl.java

```diff
@@ -95,22 +123,32 @@ public int getClassId() {
 
     @Override
     public void writeData(ObjectDataOutput out) throws IOException {
-        out.writeObject(partitionKey);
+        if (out.getVersion().isLessThan(Versions.V5_2)) {
+            out.writeObject(partitionKey);
```

Comment:
if version is less than 5.2, we are sending null predicate in this PR, this should be fixed by sending the single key in `partitionKeys`. Otherwise existing code will be broken if one upgrades 5.2. Even your hazelcast is 5.2, your cluster can be 5.1 so we must keep compatibility.